### ISSUE TITLE
Reverted #73651

### ISSUE THREAD

**Issue by vitalybuka** [llvm/llvm-project#76535]
Reverted #73651 to fix bots

**Comment by arcbbb** [comment]
@vitalybuka Could you help check the buildbot script?

Using the steps to reproduce the error, I think the build is not using the latest commit.
The LLVM version of opt should be `18.0.0git` instead of `17.0.6`
```
scratch_dir/llvm-project$ ../llvm_build0/bin/opt -version
LLVM (http://llvm.org/):
  LLVM version 17.0.6
  Optimized build.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: cascadelake
```
I think it is built before checking out `3e75dece919`
```
scratch_dir/llvm-project$ git reflog
3e75dece9195 (HEAD) HEAD@{0}: checkout: moving from 6009708b4367171ccdbf4b5905cb6a803753fe18 to 3e75dece919511e4a2edada82d783304cc14a9cd
6009708b4367 (tag: llvmorg-17.0.6, origin/release/17.x) HEAD@{1}: checkout: moving from master to llvmorg-17.0.6
```
My reproduce steps is 
```
BUILDBOT_REVISION=3e75dece919511e4a2edada82d783304cc14a9cd  llvm-zorg/zorg/buildbot/builders/sanitizers/buildbot_fast.sh
```

**Comment by arcbbb** [comment]
I figure out the error now.
```
AddressSanitizer: stack-use-after-scope on address 0x7f136cb49f40 at pc 0x55eb0c53ffc9 bp 0x7ffce6337890 sp 0x7ffce6337888
```
and I will fix it in the commit. Thanks!

**Comment by arcbbb** [comment]
Fixed and re-commit in #76793
